indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -174,6 +176,13 @@ def _validate_room_booking_limit(self, start_dt, end_dt, booking_limit_days):
         selected_period_days = (day_end_dt - day_start_dt).days
         return selected_period_days <= booking_limit_days
 
+    def _link_booking(self, booking, type_, id_):
+        if id_ and type_:
```

Comment:
0 could be a valid id :) very unlikely to happen but migrated databases to have an event and contribution with id 0.

so better to explicitly check for None here - and you could return early if type/id are missing and thus outdent the rest of this method